packages/gatsby-theme-docs/src/components/video-player-client-side.js

```diff
@@ -0,0 +1,57 @@
+import React, { useRef, useEffect } from 'react';
+import videojs from 'video.js';
+import PropTypes from 'prop-types';
+import 'video.js/dist/video-js.css';
+
+const prepareVideoOptions = (options, videoUrl, autoplay, poster) => {
+  const currOptions = options || {};
+  const sources = [{ src: videoUrl }];
+  return { ...currOptions, sources, autoplay, poster };
+};
+
+const VideoPlayer = (props) => {
+  const videoRef = useRef(null);
+  const playerRef = useRef(null);
+
+  useEffect(() => {
+    if (!playerRef.current) {
+      const videoElement = videoRef.current;
+
+      if (!videoElement) return;
+
+      playerRef.current = videojs(
+        videoElement,
+        prepareVideoOptions(
+          props.options,
+          props.videoUrl,
+          props.autoplay,
+          props.thumbnail
+        )
+      );
+    }
+  }, [props, videoRef]);
+
+  useEffect(() => {
+    const player = playerRef.current;
+    return () => {
+      if (player) {
+        player.dispose();
+        playerRef.current = null;
+      }
+    };
+  }, [playerRef]);
+
+  return (
+    <div data-vjs-player>
+      <video ref={videoRef} className="video-js vjs-big-play-centered" />
+    </div>
+  );
+};
+VideoPlayer.propTypes = {
```

Comment:
We need to think a bit better the interface of the Component.
Since everything gets passed to the videoJs in a single `options` object, I thought to implement a  list of `important` props that needs to be explicit for the editors to be able to play with them, but also accept the full option object to cater for corner cases. I implemented a override mechanism in a way that the explicit props are always overriding the option object

Reply:
I am not a fan of officially allowing direct interaction with the underlying video player library at all.   Conceptually everyhing in the docs kit is conventions based and KISS for authors.    Plus, staying able to swap the underlying library is a plus, too. 

Getting the default behavior right  (i.e. very good) through feedback is worth more to me. 

Looking at https://videojs.com/guides/options/  I'n not seeing something that I want to be different per-content.